static/app/components/searchSyntax/parser.tsx

```diff
@@ -415,6 +417,12 @@ export class TokenConverter {
     };
   };
 
+  tokenParen = (value: '(' | ')') => ({
+    ...this.defaultTokenFields,
+    type: value === '(' ? (Token.L_PAREN as const) : (Token.R_PAREN as const),
+    value,
+  });
```

Comment:
IMO have two token functions here, makes it a bit easier to understand

```suggestion
  lTokenParen = (value: '(') => ({
    ...this.defaultTokenFields,
    type: Token.L_PAREN as const
    value,
  });
  
  rTokenParen = (value: ')') => ({
    ...this.defaultTokenFields,
    type: Token.R_PAREN as const
    value,
  });
```